cmd/skaffold/app/cmd/flags.go

```diff
@@ -17,243 +17,164 @@ limitations under the License.
-		Shorthand:     "n",
-		Usage:         "Run deployments in the specified namespace",
-		Value:         &opts.Namespace,
-		DefValue:      "",
-		FlagAddMethod: "StringVar",
-		DefinedOn:     []string{"all"},
+		DefinedOn: []string{"dev", "build", "run", "debug", "deploy", "delete"},
+		Setup: func(f *pflag.FlagSet) {
+			f.StringVarP(&opts.Namespace, "namespace", "n", "", "Run deployments in the specified namespace")
+		},
 	},
 	{
-		Name:          "default-repo",
-		Shorthand:     "d",
-		Usage:         "Default repository value (overrides global config)",
-		Value:         &opts.DefaultRepo,
-		DefValue:      "",
-		FlagAddMethod: "StringVar",
-		DefinedOn:     []string{"all"},
+		DefinedOn: []string{"dev", "build", "run", "debug", "deploy", "delete"},
+		Setup: func(f *pflag.FlagSet) {
+			f.StringVarP(&opts.DefaultRepo, "default-repo", "d", "", "Default repository value (overrides global config)")
+		},
 	},
 	{
-		Name:          "cache-artifacts",
-		Usage:         "Set to true to enable caching of artifacts.",
-		Value:         &opts.CacheArtifacts,
-		DefValue:      false,
-		FlagAddMethod: "BoolVar",
-		DefinedOn:     []string{"dev", "build", "run", "debug"},
+		DefinedOn: []string{"dev", "build", "run", "debug"},
+		Setup: func(f *pflag.FlagSet) {
+			// TODO: remove dot
+			f.BoolVar(&opts.CacheArtifacts, "cache-artifacts", false, "Set to true to enable caching of artifacts")
+		},
 	},
 	{
-		Name:          "cache-file",
-		Usage:         "Specify the location of the cache file (default $HOME/.skaffold/cache)",
-		Value:         &opts.CacheFile,
-		DefValue:      "",
-		FlagAddMethod: "StringVar",
-		DefinedOn:     []string{"dev", "build", "run", "debug"},
+		DefinedOn: []string{"dev", "build", "run", "debug"},
+		Setup: func(f *pflag.FlagSet) {
+			f.StringVar(&opts.CacheFile, "cache-file", "", "Specify the location of the cache file (default $HOME/.skaffold/cache)")
+		},
 	},
 	{
-		Name:          "insecure-registry",
-		Usage:         "Target registries for built images which are not secure",
-		Value:         &opts.InsecureRegistries,
-		DefValue:      []string{},
-		FlagAddMethod: "StringSliceVar",
-		DefinedOn:     []string{"dev", "build", "run", "debug"},
+		DefinedOn: []string{"dev", "build", "run", "debug"},
+		Setup: func(f *pflag.FlagSet) {
+			f.StringSliceVar(&opts.InsecureRegistries, "insecure-registry", nil, "Target registries for built images which are not secure")
+		},
 	},
 	{
-		Name:          "enable-rpc",
-		Usage:         "Enable gRPC for exposing Skaffold events (true by default for `skaffold dev`)",
-		Value:         &opts.EnableRPC,
-		DefValue:      false,
-		FlagAddMethod: "BoolVar",
-		DefinedOn:     []string{"dev", "build", "run", "debug", "deploy"},
+		DefinedOn: []string{"dev", "build", "run", "debug", "deploy"},
+		Setup: func(f *pflag.FlagSet) {
+			f.BoolVar(&opts.EnableRPC, "enable-rpc", false, "Enable gRPC for exposing Skaffold events (true by default for `skaffold dev`)")
+		},
 	},
+	//
```

Comment:
nit: remove unnecessary comment

Reply:
I’ve got an updated version without this comment, that I couldn’t push before I took the plane :-)